test/distributed/org/apache/cassandra/distributed/test/accord/AccordCQLTestBase.java

```diff
@@ -3415,4 +3415,20 @@ public void multiPartitionUpdate() throws Exception
             }
         });
     }
+
+    @Test
+    public void userSeesInvalidRejection() throws Exception
+    {
+        var expectedType = AssertionUtils.isInstanceof(InvalidRequestException.class);
+        test("CREATE TABLE " + qualifiedAccordTableName + "(k int PRIMARY KEY, l list<int>) WITH " + transactionalMode.asCqlParam(), cluster -> {
+            String cql = "UPDATE " + qualifiedAccordTableName + " SET l[0] = 42 WHERE k=42";
+            Assertions.assertThatThrownBy(() -> cluster.coordinator(1).execute(cql, QUORUM))
+                      .is(expectedType)
+                      .hasMessage("Attempted to set an element on a list which is null");
+
+            Assertions.assertThatThrownBy(() -> cluster.coordinator(1).execute(wrapInTxn(cql), QUORUM))
+                      .is(expectedType)
+                      .hasMessage("Attempted to set an element on a list which is null");
```

Comment:
nit: It would be nice to not duplicate the error message text I guess, but we don't seem to be in the habit of doing that in `Lists`.